Userland/Libraries/LibJS/Bytecode/Pass/GenerateCFG.cpp

```diff
@@ -84,60 +87,77 @@ static void generate_cfg_for_block(BasicBlock const& current_block, PassPipeline
             auto finalizer_target = static_cast<Op::EnterUnwindContext const&>(instruction).finalizer_target();
 
             // We keep the frame alive here on the stack, to save some allocation size
-            UnwindFrame frame { nullptr, finalizer_target.has_value() ? &finalizer_target->block() : nullptr };
+            UnwindFrame frame {
+                .handler = nullptr,
+                .finalizer = finalizer_target.has_value() ? &finalizer_target->block() : nullptr
+            };
+
             unwind_frames.append(&frame);
 
+            // We enter the handler target now, to avoid it mutating the unwind stack
+            // without us noticing
+            // FIXME: There probably is a nicer way of doing this
             if (handler_target.has_value()) {
-                // We might pop from this context while in the handler, so we
-                // need to save it and return it to its rightful place
-                // FIXME: We can relax this when we are stricter about entering finally statements
                 auto saved_context = unwind_frames;
+                // FIXME: Not 100% to enter the handler from the current block
```

Comment:
"Not 100% to enter the handler" doesn't make any sense to me, please expand this comment